fuzz/fuzz_targets/archive.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![no_main]
+
+use libfuzzer_sys::fuzz_target;
+
+use std::fs::{File, OpenOptions};
+use std::io::{Cursor, Write, Read};
+use tar::{Archive, Builder, EntryType, Header};
+use tempfile::tempdir;
+use std::convert::TryInto;
+use std::str;
+
+fuzz_target!(|data: &[u8]| {
+    // Skip this iteration when data is not enough
+    if data.len() < 10 {
+        return;
+    }
+
+    // Create temp file and dir
+    let temp_dir = tempdir().unwrap();
+    let file_name = match str::from_utf8(&data[0..data.len().min(10)]) {
+        Ok(name) => name.to_string(),
+        Err(_) => "default_file_name".to_string(),
+    };
+    let dir_name = match str::from_utf8(&data[data.len().min(10)..data.len().min(20)]) {
```

Comment:
Then from here we can just reference `data` as if it was consumed.

Another way to do all of this is just to wrap `data` in a https://doc.rust-lang.org/std/io/struct.Cursor.html and call `read` or `read_exact` etc on it instead.

Reply:
Have fixed both dir_name and file_name